src/index.js

```diff
@@ -266,6 +296,13 @@ export default function ({ Plugin, types: t }) {
 
       Program: {
         enter(node, parent, scope, file) {
+          const options = getPluginOptions(file);
+          const factoryMethods = options.factoryMethods || ['React.createClass', 'createClass'];
+          this.state[optionsKey] = options;
+          this.state[cacheKey] = {
+            isCreateClassCallExpression: buildIsCreateClassCallExpression(factoryMethods),
```

Comment:
I opted to build the matchers up front instead of memoizing them off of the method name. It seemed a little simpler/more efficient in the common case.